src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,120 @@
+            Product newProduct = new Product(product.getName(), product.getPrice());
+            PreparedStatement statement = connection
+                    .prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, product.getName());
+            statement.setDouble(2, product.getPrice());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                newProduct.setId(resultSet.getLong(1));
+            }
+            return newProduct;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create product", e);
+        }
+    }
+
+    @Override
+    public Product update(Product product) {
+        String query = "UPDATE products SET name = ?, "
+                + "price = ? WHERE product_id = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, product.getName());
+            statement.setDouble(2, product.getPrice());
+            statement.setLong(3, product.getId());
+            int successfulUpdate = statement.executeUpdate();
+            if (successfulUpdate != 0) {
+                return product;
+            }
+            throw new SQLException();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t update product", e);
+        }
+    }
+
+    @Override
+    public Optional<Product> getById(Long productId) {
+        String query = "SELECT * FROM products WHERE product_id = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, productId);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Product product = getProductFromResultSet(resultSet);
+                return Optional.of(product);
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get product by id", e);
+        }
+    }
+
+    @Override
+    public List<Product> getAll() {
+        String query = "SELECT * FROM products WHERE deleted = FALSE";
+        List<Product> productList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            ResultSet resultSet = preparedStatement.executeQuery(query);
+            while (resultSet.next()) {
+                long productId = resultSet.getLong("product_id");
+                Product product = getProductFromResultSet(resultSet);
+                product.setId(productId);
```

Comment:
you've already done it inside of the method getProductFromResultSet
this code is redundant